source/utils/bpcmp/bpcmp.py

```diff
@@ -0,0 +1,143 @@
+                    print("DIFFER : VARIABLE : "+varName+" : MIN : "+str(varInfoS1[varName]["Min"])+" <> "+str(varInfoS2[varName]["Min"])+" : "+filename1+" : "+filename2)
+                    err += 1
+                if DetectMismatch(int(varInfoS1[varName]["Max"]), int(varInfoS2[varName]["Max"]), tolerance, tolperc):
+                    print("DIFFER : VARIABLE : "+varName+" : MAX : "+str(varInfoS1[varName]["Max"])+" <> "+str(varInfoS2[varName]["Max"])+" : "+filename1+" : "+filename2)
+                    err += 1
+            else:
+                #compare values
+                for step in range(varSteps):
+                    values1 = s1.read(varName, step_selection=[step, 1])
+                    values2 = s2.read(varName, step_selection=[step, 1])
+                    ret = CompareValues(values1, values2, tolerance, tolperc)
+                    if len(ret) == 0:
+                        continue
+                    err += len(ret)
+                    if verbose:
+                        for (idx, value1, value2) in ret:
+                            print("DIFFER : VARIABLE : "+varName+" : STEP : "+ str(step) +" : POSITION : "+str(idx)+" : VALUE : "+str(value1)+" <> "+str(value2)+" : "+filename1+" : "+filename2)
+                    else:
+                        print("DIFFER : VARIABLE : "+varName+" : STEP : "+ str(step) +" : VALUES_SUMMARY : "+str(len(ret))+" : "+filename1+" : "+filename2)
+
+    return err
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(prog="bpcmp", description='Utility code to compare the data within BP files', epilog='If more than two files are provided, all files are compared to the first file.')
+    parser.add_argument('bpfiles', metavar='file.bp', nargs='+',
+                    help='list of BP files to be compared')
+    parser.add_argument('-t', '--tolerance', dest='tolerance',
+                    type=float, default=0, help='tolerance for comparing values (default: 0)')
+    parser.add_argument('-p', '--perc-tolerance', dest='tolperc',
+                    type=float, default=0, help='tolerance for comparing values as percentage of values in the file (default: 0)')
+    parser.add_argument('-s', '--stats', action='store_true', help='compare stats (min/max) and not values')
+    parser.add_argument('-v', '--verbose', action='store_true', help='all different values and their indeces are printed, not just a summary')
+
+    args = parser.parse_args()
+    if len(args.bpfiles) < 2:
+        print("WARNING : At least two files need to be provided for any comparison")
+        exit()
+
+    verbose = args.verbose
+
+    diff_found = 0
+    for i in range(1, len(args.bpfiles)):
+        print("COMPARE : FILE :", args.bpfiles[0], ": FILE :", args.bpfiles[i])
+        diff_found += CompareFiles(args.bpfiles[0], args.bpfiles[i], args.tolerance, args.tolperc, args.stats)
+
+    if diff_found == 0:
+        print("DONE : No differences between the two files")
```

Comment:
between the files? all files?

Reply:
If there are more than two files provided, I am only comparing everything with the first one. Do you want me to do an all to all?

Reply:
Sorry, I just meant to fix the message.